kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/timeout/EngineTimeoutTest.kt

```diff
@@ -18,50 +17,41 @@ class EngineTimeoutTest : FunSpec() {
          TestEngineLauncher(collector)
             .withClasses(DannyDelay::class)
             .launch()
-         collector.waitForEnginesFinished()
          collector.names shouldBe listOf("a")
-         collector.result("a").asClue { result ->
-            result?.errorOrNull?.message shouldBe "Test 'a' did not complete within 1ms"
-         }
+         collector.result("a")!!.errorOrNull!!.message!! shouldBe "Test 'a' did not complete within 400ms"
       }
 
       test("timeouts should be applied by the engine to suspend inside launched coroutines") {
          val collector = CollectingTestEngineListener()
          TestEngineLauncher(collector)
             .withClasses(LarryLauncher::class)
             .launch()
-         collector.waitForEnginesFinished()
          collector.names shouldBe listOf("a")
-         collector.result("a").asClue { result ->
-            result?.errorOrNull?.message shouldBe "Test 'a' did not complete within 1ms"
-         }
+         collector.result("a")!!.errorOrNull!!.message!! shouldBe "Test 'a' did not complete within 400ms"
       }
 
       test("timeouts should be applied by the engine to blocked threads") {
          val collector = CollectingTestEngineListener()
          TestEngineLauncher(collector)
             .withClasses(BillyBlocked::class)
             .launch()
-         collector.waitForEnginesFinished()
          collector.names shouldBe listOf("a")
-         collector.result("a").asClue { result ->
-            result?.errorOrNull?.message shouldBe "sleep interrupted"
-         }
+         collector.result("a")!!.errorOrNull!!.message!! shouldBe "sleep interrupted"
```

Comment:
Could the assertions here be kept? I think that `collector.result("a").asClue { result -> }` will be more useful if the test fails than `collector.result("a")!!`

Reply:
Sure.